### PR TITLE
feat: new tabs components

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Feature

#### What is expected?

New tabs component

## Relevant logs and/or screenshots

Capture d’écran 2022-08-25 à 16 14 52

### CODE REVIEW

File: src/components/Tabs/Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  border-radius: 50%;
+  font-size: 12px;
+  background-color: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border: 1px solid ${({ theme }) => theme.colors.neutral.borderStrong};
+  margin-left: ${({ theme }) => theme.space['1']};
+  height: 24px;
+  width: 24px;
+  text-overflow: ellipsis;
+  line-height: 22px;
+`
```
Comment (matthprost) [inline src/components/Tabs/Tab.tsx:26]: It couldn't work to use component `Text` ? So we can remove all props linked to text

File: src/components/Tabs/Tab.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  text-decoration: none;
+  user-select: none;
+  touch-action: manipulation;
+  transition: color 0.2s;
+  border: none;
+  background: none;
+  border-bottom-width: 2px;
+  border-bottom-style: solid;
+  border-bottom-color: ${({ theme }) => theme.colors.neutral.borderWeak};
+  outline: none;
```
Comment (matthprost) [inline src/components/Tabs/Tab.tsx:56]: Here also we have a lot of props linked to text maybe we should use a component text inside of this component?

File: src/components/Tabs/Tab.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  &[aria-selected='true'] {
+    color: ${({ theme }) => theme.colors.primary.text};
+    border-bottom-color: ${({ theme }) => theme.colors.primary.borderWeak};
+    ${StyledCounter} {
+      background-color: ${({ theme }) => theme.colors.primary.background};
+      border-color: ${({ theme }) => theme.colors.primary.background};
+    }
+  }
+
+  &:not([aria-disabled='true']):not(:disabled) {
```
Comment (matthprost) [inline src/components/Tabs/Tab.tsx:78]:
```suggestion
  &[aria-disabled='false']:not(:disabled) {
```

?

File: src/components/Tabs/Tab.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    badge,
+    children,
+    className,
+    counter,
+    disabled = false,
+    value,
+    onClick,
+    onKeyDown,
+    tooltip,
+    ...props
```
Comment (matthprost) [inline src/components/Tabs/Tab.tsx:133]: Do we still need props spreading ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Tabs/index.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  margin-block-start: 0;
+  margin-block-end: 0;
+  padding-inline-start: 0;
+  overflow-x: scroll;
+  position: relative;
+  z-index: 0;
+  list-style-type: none;
+
+  &::after {
+    z-index: -1;
```
Comment (matthprost) [inline src/components/Tabs/index.tsx:20]: Why using z-index ? Maybe put it into `::before` ?

File: src/components/Tabs/index.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+const TabsContainer = styled.ul`
+  display: flex;
+  flex-wrap: nowrap;
+  margin-block-start: 0;
+  margin-block-end: 0;
+  padding-inline-start: 0;
+  overflow-x: scroll;
+  position: relative;
+  z-index: 0;
```
Comment (matthprost) [inline src/components/Tabs/index.tsx:16]: Please try to avoid z-index

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Tabs/Tab.tsx (hunk 4 above)
Comment (DorianMaliszewski) [reply src/components/Tabs/Tab.tsx:133]: Yes because of as prop

File: src/components/Tabs/index.tsx (hunk 5 above)
Comment (DorianMaliszewski) [reply src/components/Tabs/index.tsx:20]: Doesn't work too need z-index 😞

**Review by matthprost — CHANGES_REQUESTED** [review]
Screenshot 2022-08-26 at 10 23 59
Screenshot 2022-08-26 at 10 24 03

Seems like horizontal scroll bar is all the time displayed and not compatible with dark theme 🤔

**Comment by BlandinePelat** [comment]
- More tabs shouldn't take the width left 
- Do we have the behavior when we select a choice inside the more option?

File: src/components/Menu/index.tsx [hunk 7]
```diff
@@ -190,7 +196,11 @@ const Menu = ({
 
   // if you need dialog inside your component, use function, otherwise component is fine
   const target = isValidElement(disclosure) ? disclosure : disclosure(popover)
-  const innerRef = useRef(target) as unknown as RefObject<HTMLButtonElement>
+  const innerRef = useRef(
+    target,
+  ) as unknown as MutableRefObject<HTMLButtonElement | null>
```
Comment (QuiiBz) [inline src/components/Menu/index.tsx:201]: Why do we need to cast the whole `useRef`? I would prefer to cast only `target`

File: src/components/Tabs/Tab.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        as={computedAs}
+        aria-label={value ? `${value}` : undefined}
+        aria-selected={isSelected}
+        aria-disabled={disabled}
+        disabled={computedAs === 'button' ? disabled : undefined}
+        onClick={event => {
+          if (value) {
+            onChange(value)
+          }
+          ;(onClick as MouseEventHandler<HTMLElement>)?.(event)
```
Comment (QuiiBz) [inline src/components/Tabs/Tab.tsx:163]: `onClick` already has a type of `MouseEventHandler<HTMLElement> | undefined`, why do we need to cast?

File: src/components/Tabs/Tab.tsx [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        }}
+        onKeyDown={event => {
+          ;(onKeyDown as KeyboardEventHandler<HTMLElement>)?.(event)
```
Comment (QuiiBz) [inline src/components/Tabs/Tab.tsx:166]: Same as `onClick`

File: src/components/Tabs/index.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+import TabMenu from './TabMenu'
+import TabMenuItem from './TabMenuItem'
+import TabsContext from './TabsContext'
+
+const MenuContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  ${StyledTabButton} {
+    font-size: 14px;
```
Comment (QuiiBz) [inline src/components/Tabs/index.tsx:20]: Font size can be from theme

File: src/components/Tabs/index.tsx [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        tabsRef.current.scrollWidth - tabsRef.current.clientWidth
+          ? 'none'
+          : ''
+    }
+    if (element && displayMore) {
+      element.addEventListener('scroll', handler)
+    }
+
+    return () => {
+      if (element) element.removeEventListener('scroll', handler)
```
Comment (QuiiBz) [inline src/components/Tabs/index.tsx:124]: I think we should also check for `displayMore` as above in this `if`

**Review by QuiiBz — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**